### PR TITLE
fix: 🐛 Fix tray icon not showing/hiding the Ferdi window

### PR DESCRIPTION
**Pull request by martinbernat**
We used `app.mainWindow` at multilple places which is undefined in the newest builds due to #2049. Use `BrowserWindow.getAllWindows()` instead.

- implement toggle Ferdi window on left mouse click
- right mouse click > "Show Ferdi / Hide Ferdi" has toggle behavior as well

✅ Closes: #2147, #2054, #1955

#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉 https://github.com/sponsors/getferdi
  👉 https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.

- [ ] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [ ] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [ ] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Checklist

- [ ] My pull request is properly named
- [ ] The changes respect the code style of the project (`npm run prepare-code`)
- [ ] `npm test` passes
- [ ] I tested/previewed my changes locally

### CODE REVIEW

File: src/lib/Menu.js [hunk 1]
```diff
@@ -950,9 +950,9 @@ class FranzMenu {
         click: () => {
           this.actions.service.setActive({ serviceId: service.id });
 
-          if (isMac && i === 0) {
-            app.mainWindow.restore();
-          }
+          if (isMac && i === 0) { // feat(Mac): Open Window with Cmd+1
+            getCurrentWindow().restore();
+          }
```
Comment (vraravam) [inline src/lib/Menu.js:955]: please remove trailing spaces

File: src/lib/Tray.js [hunk 2]
```diff
@@ -28,30 +29,22 @@ export default class TrayIcon {
...
-        if (app.mainWindow.isMinimized()) {
-          app.mainWindow.restore();
-        } else if (app.mainWindow.isVisible()) {
-          app.mainWindow.hide();
-        } else {
-          app.mainWindow.show();
-          app.mainWindow.focus();
-        }
+        // Just show the window, no matter what (no toggle)
+        mainWindow.show();
```
Comment (vraravam) [inline src/lib/Tray.js:39]: How does this work if the user has chosen to start minimized eg when rebooting?

**Comment by vraravam** [comment]
@all-contributors - please add @martinbernat for code, bug

File: src/lib/Menu.js (hunk 1 above)
Comment (martinbernat) [reply src/lib/Menu.js]: sure, sorry didn't notice that :(

File: src/lib/Tray.js (hunk 2 above)
Comment (martinbernat) [reply src/lib/Tray.js:39]: This does not affect the "start minimized" option, user has to click the "Show Ferdi" item in the tray menu to see the action.

If we want to have the toggle behavior (_toggleWindow) also under this "Show Ferdi" item, we should the dynamically change the menu item from "Show Ferdi" to "Hide Ferdi" and vice versa (it is doable of course).

I thought that this is not necessary, if we have the left mouse click behavior, which is quick and straightforward. But having simple "Show Ferdi" in the menu item can be helpful form some users.

What do you think? Would you rather have the "toggling" menu item?

Comment (vraravam) [reply src/lib/Tray.js:39]: Yes - toggling would be better

**Review by vraravam — CHANGES_REQUESTED** [review]

File: src/lib/Tray.js (hunk 2 above)
Comment (martinbernat) [reply src/lib/Tray.js]: I've implemented the toggle. Menu item changes the text dynamically from "Show Ferdi" to "Hide Ferdi" and vice versa. Respecting the "Disable/Enable notifications" texts. 18n support and removing the hardcoded strings looks like my new mini project :)

**Review by vraravam — APPROVED** [review]
Works great on mac! Thanks for the contribution @martinbernat ! lgtm!

**Comment by MartinX3** [comment]
Can't wait for 5.6.4, since I like to close the window into the tray and this small bug bamboozles me a bit.

**Comment by vraravam** [comment]
though not signed, since you are able to build locally, you can always run the packaged app as the main one (that's what I do most of the time). If not, you can use the latest nightly in a couple of hours - it should have your fix!

**Comment by pulver22** [comment]
Hi @martinbernat, great work. Thank you for addressing the system tray icon but did you test it against Ubuntu (in general)? I am on 18.04 and nothing really changed on my side after installing the Nightly 7 containing your patch.

**Comment by martinbernat** [comment]
> Hi @martinbernat, great work. Thank you for addressing the system tray icon but did you test it against Ubuntu (in general)? I am on 18.04 and nothing really changed on my side after installing the Nightly 7 containing your patch.

Yes, I've tested on Ubuntu 20.10 and macOS, both worked fine. But another bug is logged against Linux Mint, and I guess that it is an issue with older Gnome (or Electron) versions. Working on it.

**Comment by shiftokay** [comment]
Hi, I'm not familiar with coding or stuffs and I just want to fix Ferdi. How do I fix this it's the same problem as written in the #2301. I can't show/hide Ferdi in the system tray, and I don't understand what I'm supposed to do to fix it.

**Comment by martinbernat** [comment]
> Hi, I'm not familiar with coding or stuffs and I just want to fix Ferdi. How do I fix this it's the same problem as written in the #2301. I can't show/hide Ferdi in the system tray, and I don't understand what I'm supposed to do to fix it.

Which version are you using and what OS? This should be already fixed in nightly builds.

**Pull request merged**